### ISSUE TITLE
LongLine rule doesnt fix long complex expression

### ISSUE THREAD

**Issue by Arrgentum** [saveourtool/diktat#1415]
### Initial code:
```kotlin
attrs.disabled = (selfRole == Role.OWNER && isSelfRecord(props.selfUserInfo, user)) || !(selfRole.isHigherOrEqualThan(Role.OWNER) || userRole.isLowerThan(selfRole))
```
Error message - ([LONG_LINE] this line is longer than allowed: max line length 180, but was 196)

### Expected code:
```kotlin
attrs.disabled = (selfRole == Role.OWNER && isSelfRecord(props.selfUserInfo, user)) || !(selfRole.isHigherOrEqualThan(Role.OWNER) || 
userRole.isLowerThan(selfRole))
```

### Observed code:

```kotlin
attrs.disabled = (selfRole == Role.OWNER && isSelfRecord(props.selfUserInfo, user)) || !(selfRole.isHigherOrEqualThan(Role.OWNER) || userRole.isLowerThan(selfRole))
```

## Environment information
* diktat version: 1.2.0

**Comment by Arrgentum** [comment]
Error in method fix in class Binary Expression - if `binary expression` contained in `parentheses` find child with type `operation references` return null
